packages/cht/test/Adaptor.test.js

```diff
@@ -0,0 +1,198 @@
+import { expect } from 'chai';
+import { enableMockClient } from '@openfn/language-common/util';
+
+import { request, get, post, put } from '../src/Adaptor.js';
+
+const testServer = enableMockClient('https://example.cht.com');
+
+describe('request', () => {
+  it('should send a GET request', async () => {
+    const data = {
+      has_permissions: true,
+    };
+
+    testServer
+      .intercept({
+        path: '/api/v1/settings',
+        method: 'GET',
+        headers: {
+          Authorization: 'Basic aGVsbG86dGhlcmU=',
+        },
+      })
+      .reply(200, data);
+
+    const state = {
+      configuration: {
+        baseUrl: 'https://example.cht.com',
+        username: 'hello',
+        password: 'there',
+      },
+      data,
+    };
+
+    const finalState = await request('GET', '/api/v1/settings')(state);
+
+    expect(finalState.data).to.eql(data);
+  });
+
+  it('should throw for a 403 response', async () => {
+    testServer
+      .intercept({
+        path: '/api/v1/contacts',
+        method: 'GET',
+      })
+      .reply(403);
+
+    const state = {
+      configuration: {
+        baseUrl: 'https://example.cht.com',
+        username: 'hello',
+        password: 'there',
+      },
+    };
+
+    const error = await request(
+      'GET',
+      '/api/v1/contacts'
+    )(state).catch(error => {
+      return error;
+    });
+    expect(error.statusMessage).to.eql('Forbidden');
+  });
+
+  it('should include basic auth', async () => {
```

Comment:
it doesn't really matter, but I would probably remove this test. It's just the same as the first get test.

Thinking about it, it would be more interesting to make sure that there's one test for each request argument  the methods, data, path and options. And we should test that each of those arguments is expanded (ie, can take a function).

But I don't think it's worth spending a lot of time on this. We're built on well-tested code here.

I'll leave it up to you. If you have an hour and would like to spend it writing extra tests, then go for it! If you're pushed for time, I'm happy to just move on.